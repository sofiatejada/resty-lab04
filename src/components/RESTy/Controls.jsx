import React from 'react';
import PropTypes from 'prop-types';

export default function Controls({ 
  data,
  method,
  url,
  handleChange, 
  handleSubmit }) {
  return (
    <form onSubmit={handleSubmit}>

      <input 
        name="url"
        aria-label="url" 
        placeholder="URL" 
        type="url"
        value={url}
        onChange={handleChange} />

      <section>
        <label htmlFor="GET">
          GET
          <input id="GET" value="GET" name="method" type="radio" 
            onChange={handleChange}/>
        </label>

        <label htmlFor="POST">
          POST
          <input id="POST" value="POST" name="method" type="radio" 
            onChange={handleChange}/>
        </label>

        <label htmlFor="PUT">
          PUT
          <input id="PUT" value="PUT" name="method" type="radio" 
            onChange={handleChange}/>
        </label>

        <label htmlFor="DELETE">
          DELETE
          <input id="DELETE" value="DELETE" name="method" type="radio"
            onChange={handleChange} />
        </label>

        <button aria-label="submit-button">Submit</button>
      </section>
      <textarea 
        aria-label="data" 
        name="data"
        value={data}
        onChange={handleChange}
        placeholder="JSON Body" />
    </form>
  );
}

Controls.propTypes = {
  data: PropTypes.string,
  method: PropTypes.string,
  url: PropTypes.string,
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
};
