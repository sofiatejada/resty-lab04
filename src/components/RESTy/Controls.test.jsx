import React from 'react';
import { render, screen } from '@testing-library/react';
import Controls from './Controls';

describe('controls', () => {
  it('displays controls', () => {
    render(
      <Controls 
        url="" data="" handleChange={() => {}} handleSubmit={() => {}} />
    );

    const controls = screen.getByLabelText('form');
    expect(controls).toMatchSnapshot();
  });
});
