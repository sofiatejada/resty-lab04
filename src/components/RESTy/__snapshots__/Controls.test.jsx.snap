// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`controls displays controls 1`] = `
<form
  aria-label="form"
>
  <input
    aria-label="url"
    name="url"
    placeholder="URL"
    type="url"
    value=""
  />
  <section>
    <label
      for="GET"
    >
      GET
      <input
        aria-label="GET"
        id="GET"
        name="method"
        type="radio"
        value="GET"
      />
    </label>
    <label
      for="POST"
    >
      POST
      <input
        id="POST"
        name="method"
        type="radio"
        value="POST"
      />
    </label>
    <label
      for="PUT"
    >
      PUT
      <input
        id="PUT"
        name="method"
        type="radio"
        value="PUT"
      />
    </label>
    <label
      for="DELETE"
    >
      DELETE
      <input
        id="DELETE"
        name="method"
        type="radio"
        value="DELETE"
      />
    </label>
    <button
      aria-label="submit-button"
    >
      Submit
    </button>
  </section>
  <textarea
    aria-label="data"
    name="data"
    placeholder="JSON Body"
  />
</form>
`;
